osh/cmd_parse.py

```diff
@@ -1342,21 +1343,26 @@ def ParseWhileUntil(self, keyword):
     # no redirects yet
     return command.WhileUntil(keyword, cond, body_node, None)
 
-  def ParseCaseArm(self):
-    # type: () -> CaseArm
-    """
-    case_item: '('? pattern ('|' pattern)* ')'
-               newline_ok command_term? trailer? ;
-
-    Looking at '(' or pattern
+  def _ParsePatWords(self, relaxed_newlines):
+    # type: (bool) -> List[word_t]
     """
-    self.lexer.PushHint(Id.Op_RParen, Id.Right_CasePat)
+    pat_words   : pat_word (newline_ok '|' newline_ok pat_word)*
+    pat_word    : WORD
 
-    left_tok = location.LeftTokenForWord(self.cur_word)  # ( or pat
+    If `relaxed_newlines` is set to False, no newlines are allowed in the
```

Comment:
It might be better to  duplicate the functions, I think it will only be a few lines, and then they can each have their own grammar